src/source/ObjcppGenerator.scala

```diff
@@ -138,6 +138,10 @@ class ObjcppGenerator(spec: Spec) extends BaseObjcGenerator(spec) {
       val objcSelf = if (i.ext.objc && i.ext.cpp) self + "CppProxy" else self
 
       if (i.ext.cpp) {
+		// Disable warnings in Clang about directly accessing the ivars of an object
+		w.wl("#if __clang__")
+        w.wl("#pragma clang diagnostic ignored \"-Wdirect-ivar-access\"")
```

Comment:
I would remove the `w.wl("#if __clang__")` `w.wl("#endif")` wrapping, else lgtm. We currently suppress the warning via an xcconfig file but having a more selective suppression would be really nice. Thanks for working on that!